src/renderer/ducks/validator/sagas.ts

```diff
@@ -342,6 +346,55 @@ function* onBeaconNodeStatusChangeUpdateValidatorState(
     }
 }
 
+export function* exportValidatorData({
+    payload,
+    meta,
+}: ReturnType<typeof exportValidator>): Generator<
+    CallEffect | SelectEffect | PutEffect | Promise<INetworkConfig | null>,
+    void,
+    IValidatorComplete & (INetworkConfig | null) & Interchange & Genesis
+> {
+    const validator = yield select(getValidator, {publicKey: meta});
+    cgLogger.log("exporting validator", validator.name);
+
+    try {
+        let db: undefined | Interchange;
+        if (validator.beaconNodes[0]) {
+            const config = (yield readBeaconChainNetwork(validator.beaconNodes[0]))?.eth2Config || mainnetConfig;
+            const slashingProtection = new CGSlashingProtection({
+                config,
+                controller: cgDbController,
+            });
+            const genesis = yield call(new CgEth2ApiClient(config, validator.beaconNodes[0]).beacon.getGenesis);
+            db = yield call(slashingProtection.exportSlashingJson, genesis.genesisValidatorsRoot, meta);
+        }
+
+        const name = validator.name
+            .toLowerCase()
+            .replace(/ /g, "_")
+            .replace(/[^a-zA-Z0-9_]/g, "");
```

Comment:
I think we are already doing something similar somewhere. Extract to utility method?

Reply:
its similar not the same